### PR TITLE
Fix #4969 - add `json_schemer` to embedded ruby gems

### LINKED ISSUE #4969: Add `json_schemer` gem
Enhancement Request
--------------
OpenStudio includes the `json-schema` gem, but it is barely maintained and [many users recommend switching to `json_schemer` gem](https://github.com/voxpupuli/json-schema/issues/423#issuecomment-634811990). Not only is the `json_schemer` gem well maintained, but it supports newer drafts of the JSON schema that are heavily utilized these days. The `json-schema` gem has no current progress towards supporting newer drafts of the schema.

For DOE's Home Energy Score project, we are unable to use the current `json-schema` gem because it doesn't support JSON Schema draft-07. This means that we can't validate JSON files submitted and has occasionally led to issues.

(Perhaps OpenStudio should consider replacing the `json-schema` gem, but that would be a breaking change.)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------

 - Fixes #4969
 - Companion PR: https://github.com/NREL/openstudio-gems/pull/64

### Pull Request Author

 - [ ] Model API Changes / Additions
 - [ ] Any new or modified fields have been implemented in the EnergyPlus ForwardTranslator (and ReverseTranslator as appropriate)
 - [ ] Model API methods are tested (in `src/model/test`)
 - [ ] EnergyPlus ForwardTranslator Tests (in `src/energyplus/Test`)
 - [ ] If a new object or method, added a test in NREL/OpenStudio-resources: Add Link
 - [ ] If needed, added VersionTranslation rules for the objects (`src/osversion/VersionTranslator.cpp`)
 - [ ] Verified that C# bindings built fine on Windows, partial classes used as needed, etc.
 - [ ] All new and existing tests passes
 - [ ] If methods have been deprecated, update rest of code to use the new methods

**Labels:**

 - [ ] If change to an IDD file, add the label `IDDChange`
 - [ ] If breaking existing API, add the label `APIChange`
 - [ ] If deemed ready, add label `Pull Request - Ready for CI` so that CI builds your PR

### Review Checklist

This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] Code Style, strip trailing whitespace, etc.
 - [ ] All related changes have been implemented: model changes, model tests, FT changes, FT tests, VersionTranslation, OS App
 - [ ] Labeling is ok
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -833,28 +833,28 @@ if(BUILD_CLI)
...
-         set(OPENSTUDIO_GEMS_ZIP_EXPECTED_MD5 "b6bf19e6958bd41be9788a7f16fd7f27")
+         set(OPENSTUDIO_GEMS_ZIP_FILENAME "openstudio3-gems-20230918-darwin.tar.gz")
+         set(OPENSTUDIO_GEMS_ZIP_EXPECTED_MD5 "71282202360c68cf9dd0f0d585d560f2")
        endif()
      else()
        if (ARCH MATCHES "arm64")
-         set(OPENSTUDIO_GEMS_ZIP_FILENAME "openstudio3-gems-20230427-linux_arm64.tar.gz")
-         set(OPENSTUDIO_GEMS_ZIP_EXPECTED_MD5 "979a8839facf2fa1c4096dcbae39ddc2")
+         set(OPENSTUDIO_GEMS_ZIP_FILENAME "openstudio3-gems-20230918-linux_arm64.tar.gz")
+         set(OPENSTUDIO_GEMS_ZIP_EXPECTED_MD5 "TODO____TDB")
```
Comment (jmarrec) [inline CMakeLists.txt:845]: I forgot I also need to build the arm64 **linux** package

Comment (jmarrec) [reply CMakeLists.txt:845]: Ok built, added to google drive, should be hosted soonish.

Comment (jmarrec) [inline CMakeLists.txt:845]:
```suggestion
         set(OPENSTUDIO_GEMS_ZIP_EXPECTED_MD5 "c28e1b84051fbae2ae0a9f7a27416951")
```

Comment (wenyikuang) [reply CMakeLists.txt]: Has been uploaded to S3 bucket. should ready to be used.

**Pull request merged**